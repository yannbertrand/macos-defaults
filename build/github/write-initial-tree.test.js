const fs = require('fs')
jest.mock('fs')

const writeInitialTree = require('./write-initial-tree')

const initialDestinationPath = './dist-initial'
const copiedFiles = ['.gitmoji-changelogrc', 'license']

describe('write-initial-tree', () => {
  beforeEach(() => {
    writeInitialTree(initialDestinationPath)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should copy some static files', () => {
    copiedFiles.forEach(file => {
      const fileContent = readFile(`${initialDestinationPath}/${file}`)
      expect(fileContent).toEqual(`copied:./templates/${file}`)
    })
  })

  it('should write an empty readme.md file using the home template', () => {
    const rootReadmeContent = readFile(`${initialDestinationPath}/readme.md`)
    expect(rootReadmeContent).toMatchSnapshot()
  })
})

const readFile = file => fs.readFakeFileSync(file, 'utf8')
