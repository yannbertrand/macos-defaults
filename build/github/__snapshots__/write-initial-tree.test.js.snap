// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`write-initial-tree should write an empty readme.md file using the home template 1`] = `
"# macos-defaults
Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it from scratch. Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/github#readme) if you want to do the same.
"
`;
